### PR TITLE
Adding adapters for MySQL and Postgres

### PR DESCRIPTION
**Pull request by volkanunsal**

### CODE REVIEW

File: lib/polo/adapters/postgres.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module Polo
+  module Adapters
+    module Postgres
+      def on_duplicate_key_update(inserts, records)
+        raise 'Not implemented for Postgres'
```
Comment (nettofarah) [inline lib/polo/adapters/postgres.rb]: Maybe we should be a little more specific here.
Let's change it to: `on_duplicate: :override is not currently supported in the PostgreSQL adapter`

File: lib/polo/adapters/postgres.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      end
+
+      def ignore_transform(inserts, records)
+        insert_and_record = inserts.zip(records)
```
Comment (nettofarah) [inline lib/polo/adapters/postgres.rb]: I like this implementation, but find it a little hard to follow because of all the gsubing and the regexes.
Can you write some high level comments for this? (in the same format as our other comments) and also include some input and ouput samples?

File: lib/polo/configuration.rb [hunk 3]
```diff
@@ -33,5 +34,9 @@ def obfuscate(*fields)
     def on_duplicate(strategy)
       @on_duplicate_strategy = strategy
     end
+
+    def set_adapter(db)
```
Comment (nettofarah) [inline lib/polo/configuration.rb]: I don't think we're using this anywhere.
Probably worth removing this method.

File: lib/polo/sql_translator.rb [hunk 4]
```diff
@@ -7,47 +9,35 @@ class SqlTranslator
...
-      records = Array.wrap(@record)
 
-      sqls = records.map do |record|
-        raw_sql(record)
-      end
-
-      if @configuration.on_duplicate_strategy == :ignore
-        sqls = ignore_transform(sqls)
+      case @configuration.adapter
+      when :mysql then self.extend(Polo::Adapters::MySQL)
```
Comment (nettofarah) [inline lib/polo/sql_translator.rb]: Not sure I like this extend pattern much.
We could simplify things and pass in the adapter as an object in the constructor.
e.g. `def initialize(object, configuration=Configuration.new, adapter)`

That would simplify our tests a lot too.

File: lib/polo/sql_translator.rb [hunk 5]
```diff
@@ -7,47 +9,35 @@ class SqlTranslator
...
+      else
+        raise "Unknown SQL adapter: #{@configuration.adapter}"
       end
+    end
 
-      if @configuration.on_duplicate_strategy == :override
-        sqls = on_duplicate_key_update(sqls, records)
+    def to_sql
+      case @configuration.on_duplicate_strategy
+      when :ignore then ignore_transform(inserts, records)
```
Comment (nettofarah) [inline lib/polo/sql_translator.rb]: as mentioned above.. this would become `@adapter.ignore_transform(inserts, records)`

File: polo.gemspec [hunk 6]
```diff
@@ -25,6 +25,7 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "bundler", "~> 1.10"
   spec.add_development_dependency "rake", "~> 10.0"
   spec.add_development_dependency "rspec"
+  spec.add_development_dependency "pry-rails"
```
Comment (nettofarah) [inline polo.gemspec]: not sure we need this as a dev dependency. Let's take it out for now.

File: spec/adapters/mysql_spec.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
+require 'spec_helper'
+
+describe Polo::Adapters::MySQL do
+  include Polo::Adapters::MySQL
```
Comment (nettofarah) [inline spec/adapters/mysql_spec.rb]: Remove this after we turn `Polo::Adapters::MySQL` into an object.

File: spec/adapters/mysql_spec.rb [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+  end
+
+  before(:all) do
+    TestData.create_netto
+  end
+
+  let(:translator) { Polo::SqlTranslator.new(netto, Polo::Configuration.new(adapter: :mysql)) }
+
+  describe '#ignore_transform' do
+    it do
```
Comment (nettofarah) [inline spec/adapters/mysql_spec.rb]: let's make sure we add a title for this test.
`it 'appends the IGNORE command after INSERTs'`

File: spec/adapters/mysql_spec.rb [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      records = translator.records
+      inserts = translator.inserts
+      translated_sql = ignore_transform(inserts, records)
+      expect(translated_sql).to eq(insert_netto)
+    end
+  end
+
+
+  describe '#on_duplicate_key_update' do
+    it do
```
Comment (nettofarah) [inline spec/adapters/mysql_spec.rb]: `it 'appends ON DUPLICATE KEY UPDATE with all values to the current INSERT statement'`

File: spec/adapters/postgres_spec.rb [hunk 10]
```diff
@@ -0,0 +1,26 @@
+require 'spec_helper'
+
+describe Polo::Adapters::Postgres do
+  include Polo::Adapters::Postgres
```
Comment (nettofarah) [inline spec/adapters/postgres_spec.rb]: same as the other adapter.

File: spec/adapters/postgres_spec.rb [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  before(:all) do
+    TestData.create_netto
+  end
+
+  let(:translator) { Polo::SqlTranslator.new(netto, Polo::Configuration.new(adapter: :postgres)) }
+
+  describe '#ignore_transform' do
+    it do
```
Comment (nettofarah) [inline spec/adapters/postgres_spec.rb]: I don't really get what's going on here 😅
So I think a good title for this test would go a long way.

File: spec/sql_translator_spec.rb [hunk 12]
```diff
@@ -21,25 +21,4 @@
     recipe_to_sql = Polo::SqlTranslator.new(recipe).to_sql.first
     expect(recipe_to_sql).to include(%q{'{"quality":"ok"}'}) # JSON, not YAML
```
Comment (nettofarah) [inline spec/sql_translator_spec.rb]: it would probably make sense to write some simple tests here just to guarantee that we're sending calls our strategy object. Nothing too fancy, but just to keep things consistent.

**Comment by nettofarah** [comment]
@volkanunsal awesome PR!
I left some comments just to make sure we can keep the code consistent with the rest of the codebase.

Please make sure to squash all commits before and write a commit message in this format: http://chris.beams.io/posts/git-commit/

We should be good to merge after you're done making the changes and squashing the commits.

**Comment by volkanunsal** [comment]
@nettofarah Thanks for the awesome line notes. I'm going to go through them later today. :+1:

**Comment by volkanunsal** [comment]
Sorry for the delay, but took care of most of the items on the list. Take a look.

**Comment by nettofarah** [comment]
This looks awesome!
I'll take an extra look tomorrow, but everything looks good to me.

Congrats on your first PR to Polo :)

**Comment by volkanunsal** [comment]
Ping!

**Comment by nettofarah** [comment]
Yay! Sorry for the delay.
I'll draft a release soon :)

**Comment by nettofarah** [comment]
hey, @volkanunsal. would you mind writing a comment here? https://github.com/IFTTT/polo/issues/19
